src/Lean/Meta/Tactic/Simp/Simproc.lean

```diff
@@ -172,15 +172,12 @@ def addSEvalprocBuiltinAttr (declName : Name) (post : Bool) (proc : Sum Simproc
 
 def Simprocs.add (s : Simprocs) (declName : Name) (post : Bool) : CoreM Simprocs := do
   let proc ←
-    try
+    if (← isBuiltinSimproc declName) then
```

Comment:
@leodemoura By moving `isBuiltinSimproc` to the front, we avoid a potentially expensive negative environment lookup. In theory, a builtin simproc can conflict with an environment declaration, in which case we would now select the former instead of the latter. But this strikes me as a very fringe case. Do you think we should we be concerned about it?

Reply:
It is fine. Let's merge the PR.